src/main/scala/org/bitcoins/core/crypto/TransactionSignatureSerializer.scala

```diff
@@ -74,22 +74,21 @@ trait TransactionSignatureSerializer extends RawBitcoinSerializerHelper with Bit
     val updatedInputs = for {
       (input,index) <- inputSigsRemoved.zipWithIndex
     } yield {
-        if (UInt32(index) == inputIndex) inputWithConnectedScript
+        if (UInt32(index) == inputIndex) {
+          inputWithConnectedScript
+        }
         else input
       }
 
     val txWithInputSigsRemoved = Transaction(spendingTransaction,UpdateTransactionInputs(updatedInputs))
-
-    //just need to add the hash type and hash the tx
-    val sigHashBytes : List[Byte] = List(0x00.toByte, 0x00.toByte, 0x00.toByte, hashType.byte).reverse
-
+    val sigHashBytes : List[Byte] = hashType.num.bytes.reverse.toList
```

Comment:
I think the we can get rid of the type annotation ` : List[Byte]` and then remove the `.toList` call at the very end of this